core/src/main/java/org/apache/accumulo/core/iterators/SortedKeyValueIterator.java

```diff
@@ -147,4 +147,13 @@
    *              if not supported.
    */
   SortedKeyValueIterator<K,V> deepCopy(IteratorEnvironment env);
+
+  /**
+   * Closes the Iterator. This should be overridden by the implementing class that has access to <tt>SortedKeyValueIterator source</tt> provided in the
```

Comment:
Thanks for the link but I think a "SHOULD" is the most appropriate. We don't want to say MUST since there may not be anything to close. 

I think the ambiguity stems from the flexibility we have with the Iterators.  Look at the doc for init for instance: "Initializes the iterator. Data should not be read from the source in this method."  Do we want to make them more rigid?

I don't mention Exceptions because I wasn't sure what to say. Sometimes we eat, log and continue and sometimes we throw.  Do we have best practices for handling exceptions?

I can change the "Typically" part to be a little more specific.